src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```diff
@@ -109,7 +119,16 @@ public static void waitForPipelineToTransitionToStatus(String pipelineStatus, lo
    * Wait till the Pipeline's status changes (from Running) to either Succeeded, Failed or Stopped within the
    * Timeout: {@link ConstantsUtil#PIPELINE_RUN_TIMEOUT_SECONDS}
    */
-  public static void waitTillPipelineRunCompletes() {
+  public static void waitTillPipelineRunCompletes() throws InterruptedException, IOException {
+    // Adding a page refresh in case tests are running on CDF to update the pipeline status.
+    if (Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONCDF)) ||
+      Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONHDF))) {
+      do {
+        PageHelper.refreshCurrentPage();
+        SeleniumDriver.getWaitDriver(ConstantsUtil.PAGE_LOAD_TIMEOUT_SECONDS);
+      } while (isStarting() || isRunning() || isProvisioning());
+    }
+
     SeleniumDriver.getWaitDriver(ConstantsUtil.PIPELINE_RUN_TIMEOUT_SECONDS).until(ExpectedConditions.or(
       ExpectedConditions.visibilityOf(CdfPipelineRunLocators.succeededStatus),
```

Comment:
This wait seems unneeded based on the above retry which will only stop when the pipeline is in a terminal state

Reply:
Agree, The purpose of this wait here was if the pipeline is in Running status for more than ```15 minutes```  we should mark that pipeline as failed to avoid a very long Build execution time. 

Now after the above refresh logic, the do-while loop keep on refreshing the page until the pipeline is in running state which is a problem what if pipeline takes 50-60 mins. 

So added a logic in the do-while to break the loop based if the pipeline is in running state for more that ```900second or 15 mins``` also changed the ```PIPELINE_RUN_TIMEOUT_SECONDS = 900second``` to ```IMPLICIT_TIMEOUT_SECONDS = 30second```

(https://github.com/cdapio/cdap-e2e-tests/pull/197/commits/44be5d5c5c9bf2c3ee877d3eb7dd3aa43c414929#diff-cfa45894e4592a5d6ed32f485ed2594ae01b9a3b154e60820a599164e0b52e80R122)